### PR TITLE
fix: Show validation error message on facility cover image upload (#8625)

### LINKED ISSUE #8625: Show validation error message on facility cover image upload
**Describe the bug**

When we are trying to upload a cover image (attached with this issue) to a facility, the frontend is throwing the "something went wrong" error

Note: When you capture image using the camera component in the cover image module, its working

**To Reproduce**

1. Go to the facility tab
2. click on any facility card
3. try to upload a facility cover image

**Expected behavior**

- [ ] when we upload a facility cover image, it should work properly
- [ ] Rather than "something went wrong" , proper error's should be thrown incase of failure

**Screenshots**

image

### PR DESCRIPTION
**Pull request by syedfardeenjeelani**
## Proposed Changes

- Fixes #8625
- Added frontend validation for image dimensions (width and height) before uploading the facility cover image.
- Displayed a specific error message when the image exceeds the allowed 1024x1024 pixel size.

**Video Demo**:  
https://github.com/user-attachments/assets/f0c8d680-8885-4856-a2ee-77fbebd27964

@ohcnetwork/care-fe-code-reviewers  
@nihal467  

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [x] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [x] Request for Peer Reviews.
- [ ] Completion of QA.

### CODE REVIEW

**Review by rithviknishad — CHANGES_REQUESTED** [review]
Can we instead show the error from the backend as is as mentioned in issue comment instead of duplicating the validations and checks from client too?

**Comment by syedfardeenjeelani** [comment]
> Can we instead show the error from the backend as is as mentioned in issue comment instead of duplicating the validations and checks from client too?

sure

**Comment by syedfardeenjeelani** [comment]
> Can we instead show the error from the backend as is as mentioned in issue comment instead of duplicating the validations and checks from client too?

here is the demo of the video



https://github.com/user-attachments/assets/270a6b38-e85d-417e-ab4b-d15e6203de6a

File: src/Components/Facility/CoverImageEditModal.tsx [hunk 1]
```diff
@@ -130,6 +130,7 @@ const CoverImageEditModal = ({
           "Bearer " + localStorage.getItem(LocalStorageKeys.accessToken),
       },
       async (xhr: XMLHttpRequest) => {
+        const errorResponse = JSON.parse(xhr.responseText);
```
Comment (rithviknishad) [inline src/Components/Facility/CoverImageEditModal.tsx]: can we parse this only in the else block?

(i.e. the status is non 2xx)


```suggestion
        const error = JSON.parse(xhr.responseText);
```

Comment (syedfardeenjeelani) [reply src/Components/Facility/CoverImageEditModal.tsx:133]: hello @rithviknishad  should i make it like this 
 else {
    try {
      const error = JSON.parse(xhr.responseText); 
      Notification.Error({
        msg: errorResponse?.cover_image?.join(' '),
      });
    } 
    /// this is for if parsing err fails 
    catch (e) {
      Notification.Error({
        msg: "Something went wrong!",
      });
    }

**Review by rithviknishad — APPROVED** [review]

**Review by rithviknishad — APPROVED** [review]

**Comment by syedfardeenjeelani** [comment]
@rithviknishad how to fix these errors

**Comment by nihal467** [comment]
LGTM

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**